integration/spark2/src/main/scala/org/apache/spark/sql/optimizer/CarbonLateDecodeRule.scala

```diff
@@ -697,12 +697,19 @@ class CarbonLateDecodeRule extends Rule[LogicalPlan] with PredicateHelper {
         marker.pushBinaryMarker(allAttrsNotDecode)
         u
       case p: Project if relations.nonEmpty =>
-        val prExps = p.projectList.map { prExp =>
-          prExp.transform {
-            case attr: AttributeReference =>
-              updateDataType(attr, attrMap, allAttrsNotDecode, aliasMap)
-          }
-        }.asInstanceOf[Seq[NamedExpression]]
+        val prExps = if (p.projectList.toString().contains("substring")) {
```

Comment:
It is sort of a hack, what if some other udf is used or the column name contains the substring part.

Reply:
yeah, I too thought about it. couldn't find other ways to fix. Suggestions are welcome.

Reply:
ok. Handled differently now. please checl